### ISSUE TITLE
Missing braille tables in the Braille NVDA Settings

### ISSUE THREAD

**Issue by DrSooom** [nvaccess/nvda#11263]
### Menu path:
NVDA » Preferences » Settings... » Braille
### Actual behavior:
The entries for "de-g0-bidi.utb", "de-g1-bidi.ctb" and "uk-comp.utb" are missing in the GUI.
### Expected behavior:
Quote from [my comment in PR #10832](https://github.com/nvaccess/nvda/pull/10832#issuecomment-593723179):
> I suggest to add "(improved back-translation, experimental)" to the two new German entries.

[Update 2020-06-18 08:55 UTC+2]: See [@Futyn-Maker's comment in PR #11221](https://github.com/nvaccess/nvda/pull/11221#issuecomment-645368497). Therefor I expanded the actual behavior as well as the upcoming workaround section. [/Update]

See also [my comment in issue #10912](https://github.com/nvaccess/nvda/issues/10912#issuecomment-606308008). Recently I found a better workaround as I once described in the linked comment.
#### Workaround for NVDA 2020.2 Beta 1:
1. If you never changed the braille output table before, then select now another than the default braille output table via the Braille NVDA Settings and save the configuration.
2. Open "nvda.ini" in Notepad++.
3. Move to the line starting with "translationTable" for the braille output table.
4. Enter "de-g0-bidi.utb", "de-g1-bidi.ctb" or "uk-comp.utb" here instead of e.g. "de-de-comp8.ctb".
5. Restart NVDA without letting NVDA save the config on exit.

Notes: The corresponding combobox in the Settings GUI is empty after this change. Changing the "inputTable" line for the braille input to one of the above mentioned files throws an error and "en-ueb-g1.ctb" could be used instead (not tested).
#### NVDA installed/portable/running from source:
Portable
#### NVDA version:
2020.2 Beta 1

CC: @leonardder and @Andre9642

**Comment by feerrenrut** [comment]
Can you expand on what these two files are. Also, it would be good to expand this issue to address https://github.com/nvaccess/nvda/pull/11221#issuecomment-645368497

>I understand that I've been bit late, but You've forgotten to include the new Ukrainian computer braille table. If it is possible before NVDA 2020.2 release, I'll very glad.

>The table was included in Liblouis 3.14 and called "uk-comp.utb"

**Comment by leonardder** [comment]
I'm a bit confused about those new German tables. How experimental are they really?

note that we once choose to replace `en-us-comp8.ctb` by `en-us-comp8-ext.utb` because the latter was more complete and extended on the former, so it wouldn't do any harm when it was replaced.

What are the intentions of these German tables? Are they meant to replace the old tables? Are their file names persistent, i.e. meant to stay even when the tables are no longer experimental?

**Comment by Adriani90** [comment]
@leonardder they are improving back translation of minus sign and other signs such as apostrphe, tilde and so on. I have marked the author of those tables on the coresponding pull request, let's see. But these german bidi tables seem to be a newer version of the G0 and G1 tables with some corrections, so I guess there is nothing that should be broken if these tables replace the older ones.

**Comment by DrSooom** [comment]
> […], so I guess there is nothing that should be broken if these tables replace the older ones.

@Adriani90: I'm not so sure here because "de-g0.utb" is included in "de-g2.ctb" at the beginning in this file. Therefore I prefer to add two new entries in the GUI instead of replacing the older German braille tables right now.

**Comment by lukaszgo1** [comment]
@DrSooom I believe you are a braille user and German is your mother tongue - right? Have you tested these new German tables specifically? also cc @MarcoZehe who I believe have created issues about problems in German braille translation in the past.

**Comment by DrSooom** [comment]
I'm primary using the 8-dot German braille table, so I'm sadly not familiar enough with 6-dot German Grade 0 and Grade 1 to test the new braille tables here in a useful way. Sometimes I still have to take a look at the official documentation ([Das System der deutschen Brailleschrift (2018)](http://www.bskdl.org/textschrift.html)) because in the meantime I forgot several rules, as I'm using 6-dot braille too seldom. Therefore I didn't give a statement regarding the quality of the new German braille tables, because I'm the wrong person for it due to my limited knowledge about their rules. Sorry, that I can't help you here further.

**Comment by Adriani90** [comment]
Since we have enough input from the autor of these tables who tested them in practice, I have no doubt of any big issues here. G2 would remain as it is, so if G0 is included in G2 there is no issue when replacing the old table. There could be an additional combo box in NVDA's settings called bidirectional tables, which display only tables which allow both ways of translation. Then the older and the newer tables could co exist but in this case the old german G0 and G1 tables have major issues in back and also forward translation so they can be deleted in my view. See the coresponding pull request for more information.

**Comment by DrSooom** [comment]
@Adriani90: Perhaps I wasn't clear enough [here](https://github.com/nvaccess/nvda/issues/11263#issuecomment-645822766). The content of "de-g0.utb" wasn't copied to "de-g2.ctb", it is included by using the "include" opcode. So if you delete "de-g0.utb", "de-g2.ctb" will no longer work.
